### ISSUE TITLE
Prometheus: Type/version config gets stale

### ISSUE THREAD

**Issue by gtk-grafana** [grafana/grafana#71239]
The code in https://github.com/grafana/grafana/pull/57554/files#diff-508952d4a21178935a79dfdbb0985b85d6f31b07b1fe4e12bb9c2061a2d3e142R71-R75, makes sure that the version being matched is within a patch version, which helps prevent false positives, otherwise selecting the wrong datasource type could return a valid version. 

I patched this with new versions in https://github.com/grafana/grafana/issues/71155, but the new versions are a temporary fix, and not great for folks using grafana 9.x but prometheus 2.41+

This results in the version detection not working when the returned version is higher then what we've got defined, meaning we're stuck in a sisyphean task of updating lists of versions.

Can we either come up with a better way of validating versions, or give up and embrace false positives when people switch between datasources (the max Mimir version being returned for a prometheus version, example 2.40.0 prometheus selecting 2.3.0 mimir).

TL;DR:
How can we prepare version detect functionality for future data source versions?
